src/combination.jl

```diff
@@ -0,0 +1,171 @@
+    return par
+end
+
+function BlissCombination(p1::Array{Float64, 2}, p2::Array{Float64, 2}, n::Int)
+    """ A function to calculate Bliss independence for drug combination assuming
+    the two drugs hit different pathways and they effect independently. """
+
+    param1 = ParamForBliss(p1, n)
+    param2 = ParamForBliss(p2, n)
+    """ For 8x8 combination of drug concentrations for G1 progression rate, G2 progression rate, and death rates in G1 and G2, respectively. """
+    combined = zeros(n, n, 4)
+    for j = 1:n
+        for k = 1:n
+            combined[j, k, 1:2] .= -(param1[1:2, j] .+ param2[1:2, k] .- param1[1:2, j] .* param2[1:2, k]) .+ p1[1:2, 1, 1]
+            combined[j, k, 3:4] .= -(param1[3:4, j] .+ param2[3:4, k])
+        end
+    end
+    @assert(all(combined .>= 0.0))
+    return combined
+end
+
+""" To output the full ODE params for plotting the cell number. """
+function fullCombinationParam(origP1::Array{Float64, 2}, origP2::Array{Float64, 2}, origFullParam::Array{Float64, 3}, n::Int)
+    """ Here we assume the base is origP1, and we just want to get the params of EC50 from origP2. """
+    combined = BlissCombination(origP1, origP2, n)
+    fullparam = zeros(9, n, n)
+    fullparam[5:9, :, :] .= origFullParam[5:9, 1, 1]
+    for i = 1:4
+        fullparam[i, :, :] .= combined[:, :, i]
+    end
+    return fullparam
+end
+
+""" Function unit to plot drug effects before and after combination. """
+function plotunitCombin(conc::Array{Float64, 1}, gemc::Array{Float64, 1}, titles, combin::Array{Float64, 1})
+    concs = log.(conc)
+    plot(concs, gemc, ylabel = titles, label = "taxol alone", legendfontsize = 7, lw = 3, fg_legend = :transparent, shape = :circle, color = :purple)
+    plot!(concs, combin, label = "taxol w/ 5nM gemc.", lw = 3, shape = :circle, color = :green)
+end
+
+""" Function to plot all of the drug effects before and after drug combination. """
+function plotEffectsCombin(concs::Array{Float64, 2}, gemc::Array{Float64, 2}, combin::Array{Float64, 3})
+    titles = ["G1 prog. rate", "G2 prog. rate", "G1 death rate", "G2 death rate"]
+    pl = [plotunitCombin(concs[:, 4], gemc[i, :], titles[i], combin[:, 4, i]) for i = 1:4]
+    plot(pl..., layout = (2, 2))
+    plot!(size = (800, 500), margin = 0.4cm, dpi = 150)
+end
+
+function plotNumcells(drugB::Array{Float64, 2}, combination::Array{Float64, 2}, concDrugB::Array{Float64, 1}, g0::Float64, n::Int)
+    numscomb = zeros(n)
+    nums = zeros(n)
+    for j = 1:n
+        numscomb[j] = numcells(combination[:, j], g0, 96)
+        nums[j] = numcells(drugB[:, j], g0, 96)
+    end
+    plot(log.(concDrugB), numscomb, label = "pac + gemc", legendfontsize = 7, lw = 3, fg_legend = :transparent, shape = :circle, color = :purple)
+    plot!(log.(concDrugB), nums, label = "pac", lw = 3, xlabel = "log drug concentration", ylabel = "cell #", shape = :circle, color = :green)
+    plot!(dpi = 150)
+end
+
+function helperPlot(concd1, named1, concd2, named2, numscomb)
```

Comment:
Why is all this code moved? If you want to do it for cleanliness, please do it in a separate PR. The change you're suggesting for timed treatment will likely be complicated, so it will be helpful to have a clean diff.

Reply:
Sure, I'll put it back and in a new branch will clean up.

Reply:
Sorry, I will remove it and clean up in a new branch.